webui/src/pages/auth/users/index.jsx

```diff
@@ -1,51 +1,129 @@
-import React, {createContext, useCallback, useEffect, useState} from "react";
-import {Outlet} from "react-router-dom";
-import { useOutletContext } from "react-router-dom";
+import React, {createContext, useCallback, useEffect, useMemo, useState} from "react";
+import {Outlet, useOutletContext} from "react-router-dom";
 
 import Button from "react-bootstrap/Button";
+import Form from "react-bootstrap/Form";
 
-import {useAPI, useAPIWithPagination} from "../../../lib/hooks/api";
+import {useAPI} from "../../../lib/hooks/api";
 import {auth} from "../../../lib/api";
 import useUser from "../../../lib/hooks/user";
 import {ConfirmationButton} from "../../../lib/components/modals";
 import {EntityActionModal} from "../../../lib/components/auth/forms";
 import {Paginator} from "../../../lib/components/pagination";
-import {useRouter} from "../../../lib/hooks/router";
 import {Link} from "../../../lib/components/nav";
 import {
     ActionGroup,
     ActionsBar,
+    AlertError,
     Checkbox,
     DataTable,
-    AlertError,
     FormattedDate,
     Loading,
     RefreshButton
 } from "../../../lib/components/controls";
 import validator from "validator/es";
-import { disallowPercentSign, INVALID_USER_NAME_ERROR_MESSAGE } from "../validation";
-import { resolveUserDisplayName } from "../../../lib/utils";
+import {disallowPercentSign, INVALID_USER_NAME_ERROR_MESSAGE} from "../validation";
+import {resolveUserDisplayName} from "../../../lib/utils";
+import {SearchIcon} from "@primer/octicons-react";
+import InputGroup from "react-bootstrap/InputGroup";
+import {allUsersFromLakeFS} from "../../../lib/components/auth/users";
+import {useRouter} from "../../../lib/hooks/router";
 
+const DEFAULT_DEBOUNCE_MS = 300;
+const DEFAULT_LISTING_AMOUNT = 100;
+const DECIMAL_RADIX = 10;
 const USER_NOT_FOUND = "unknown";
 export const GetUserEmailByIdContext = createContext();
 
 
-const UsersContainer = ({nextPage, refresh, setRefresh, error, loading, userListResults}) => {
+const UsersContainer = ({ refresh, setRefresh, allUsers, loading, error }) => {
     const { user } = useUser();
     const currentUser = user;
 
     const router = useRouter();
-    const after = (router.query.after) ? router.query.after : "";
     const [selected, setSelected] = useState([]);
     const [deleteError, setDeleteError] = useState(null);
     const [showCreate, setShowCreate] = useState(false);
     const [showInvite, setShowInvite] = useState(false);
-    
-    
 
-    useEffect(() => { setSelected([]); }, [refresh, after]);
+    const routerSearchPrefix = (router.query.searchPrefix) ? router.query.searchPrefix : "";
```

Comment:
```suggestion
    const routerSearchPrefix = router.query.searchPrefix ? router.query.searchPrefix : "";
```

Reply:
Nit, of course: The `()` are redundant, here and in the following lines as well.